e2e-tests/batch-mode-tests.sh

```diff
@@ -1,160 +1,176 @@
+      --max-time 20 http://localhost:8099/openmrs/ws/fhir2/R4/Patient 2>/dev/null | cut -d ";" -f 1)
     ((openmrs_start_wait_time += 1))
     if [[ ${openmrs_start_wait_time} == 20 ]]; then
-      echo "${print_prefix} TERMINATING TEST AS OPENMRS TOOK TOO LONG TO START"
+      print_message "TERMINATING TEST AS OPENMRS TOOK TOO LONG TO START"
       exit 1
     fi
   done
-  echo "${print_prefix} OPENMRS SERVER STARTED SUCCESSFULLY"
+  print_message "OPENMRS SERVER STARTED SUCCESSFULLY"
 
   docker-compose -f sink-compose.yml up -d
   fhir_server_start_wait_time=0
   fhir_server_status_code=$(curl -o /dev/null --head -w "%{http_code}" -L -X GET -u hapi:hapi --connect-timeout 5 \
     --max-time 20 http://localhost:8098/fhir/Observation 2>/dev/null)
   until [[ ${fhir_server_status_code} -eq 200 ]]; do
     sleep 1s
-    echo "${print_prefix} WAITING FOR FHIR SERVER TO START"
+    print_message "WAITING FOR FHIR SERVER TO START"
     fhir_server_status_code=$(curl -o /dev/null --head -w "%{http_code}" -L -X GET -u hapi:hapi --connect-timeout 5 \
       --max-time 20 http://localhost:8098/fhir/Observation 2>/dev/null)
     ((fhir_server_start_wait_time += 1))
     if [[ fhir_server_start_wait_time == 10 ]]; then
-      echo "${print_prefix} TERMINATING AS FHIR SERVER TOOK TOO LONG TO START"
+      print_message "TERMINATING AS FHIR SERVER TOOK TOO LONG TO START"
       exit 1
     fi
   done
-  echo "${print_prefix} FHIR SERVER STARTED SUCCESSFULLY"
+  print_message "FHIR SERVER STARTED SUCCESSFULLY"
 
-  test_dir_fhir=$(mktemp -d -t analytics_tests__XXXXXX_FHIRSEARCH)
-  test_dir_jdbc=$(mktemp -d -t analytics_tests__XXXXXX_JDBC)
+  TEST_DIR_FHIR=$(mktemp -d -t analytics_tests__XXXXXX_FHIRSEARCH)
+  TEST_DIR_JDBC=$(mktemp -d -t analytics_tests__XXXXXX_JDBC)
+  JDBC_SETTINGS="--jdbcModeEnabled=true --jdbcUrl=jdbc:mysql://localhost:3306/openmrs --dbUser=mysqluser \
+                 --dbPassword=mysqlpw --jdbcMaxPoolSize=50 --jdbcDriverClass=com.mysql.cj.jdbc.Driver"
 }
 
+
+#######################################
+## Function that tests sinking to parquet files
+## and compares output to what is in openmrs server
+# Arguments:
+#   extra flags for Dexex.args
+#   file path
+#   the mode to test: FHIR Search vs JDBC
+#######################################
 function test_parquet_sink() {
+  local command=(mvn exec:java -pl batch "-Dexec.args=--openmrsServerUrl=http://localhost:8099/openmrs \
+        --openmrsUserName=admin --openmrsPassword=Admin123 \
+        --searchList=Patient,Encounter,Observation --batchSize=20 $1")
   local test_dir=$2
   local mode=$3
-  echo "${print_prefix} PARQUET FILES WILL BE WRITTEN INTO $test_dir DIRECTORY"
-  echo "${print_prefix} RUNNING BATCH MODE WITH PARQUET SYNC FOR $2 MODE"
-  echo "${print_prefix}  $1 "
-
-  $1
+  print_message "PARQUET FILES WILL BE WRITTEN INTO ${test_dir} DIRECTORY"
+  print_message "RUNNING BATCH MODE WITH PARQUET SYNC FOR $2 MODE"
+  print_message " ${command[*]}"
+  "${command[@]}"
```

Comment:
Out curiosity: Given the way that we are using these functions (and what parameters we pass to them), is there a difference between using `command[@]` or `command[*]` here? It seems to me it only makes a difference if some positional arguments have spaces in them (which I think is not the case for us).
No need to change anything, I was just curious why these two lines are not the same `command`.

Reply:
I have Shell Check in Intellij and it didn't like `" ${command[@]} "` due to [Argument mixes string and array. Use * or separate argument.](https://github.com/koalaman/shellcheck/wiki/SC2145 )